core/src/main/java/io/grpc/Deadline.java

```diff
@@ -90,7 +88,7 @@ private Deadline(Ticker ticker, long baseInstant, long offset,
    */
   public boolean isExpired() {
     if (!expired) {
-      if (deadlineNanos <= ticker.read()) {
+      if (deadlineNanos - ticker.read() <= 0) {
```

Comment:
nanoTime() may return Long.MAX_VALUE right now. If I call it again in another second I will get a negative value. That is part of "normal" operation for nanoTime.

In nanoTime's documentation:
> one should use t1 - t0 < 0, not t1 < t0, because of the possibility of numerical overflow.